controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -404,10 +408,34 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 					"test.appstudio.openshift.io/scenario":  integrationTestScenario.Name,
 				},
 			}
-			Eventually(func() bool {
-				err := k8sClient.List(adapter.context, integrationPipelineRuns, opts...)
-				return len(integrationPipelineRuns.Items) > 0 && err == nil
-			}, time.Second*10).Should(BeTrue())
+			Eventually(func() error {
+				if err := k8sClient.List(adapter.context, integrationPipelineRuns, opts...); err != nil {
+					return err
+				}
+
+				if expected, got := 1, len(integrationPipelineRuns.Items); expected != got {
+					return fmt.Errorf("found %d PipelineRuns, expected: %d", expected, got)
+				}
+
+				ipr := integrationPipelineRuns.Items[0]
+				annotations := map[string]string{
+					gitops.PipelineAsCodeInstallationIDAnnotation: "123",
+					"build.appstudio.redhat.com/commit_sha":       "6c65b2fcaea3e1a0a92476c8b5dc89e92a85f025",
+					"test.appstudio.openshift.io/kind":            "kind",
+				}
+
+				if expected, got := len(annotations), len(ipr.Annotations); expected != got {
+					return fmt.Errorf("found %d annotations on the PipelineRun, expected: %d", expected, got)
+				}
+
+				for expectedKey, expectedValue := range annotations {
+					if got := ipr.Annotations[expectedKey]; expectedValue != got {
+						return fmt.Errorf("annotation %q on the PipelineRun, expected: %q, got: %q", expectedKey, expectedValue, got)
+					}
+				}
```

Comment:
Can be this extracted out of eventually block, using standard ginkgo calls Expect() ?